### ISSUE TITLE
[Question] Handheld Weapon Clipping

### ISSUE THREAD

**Issue by Dwarf-LordPangolin** [tsunamayo/Starship-EVO#2217]
It is currently possible to clip bricks placed on the handheld weapons. I just wanted to ask if this is intentional (kinda hoping it is because it lets us build sleeker guns). Thanks!

**Comment by Aiwii** [comment]
I agree with dwarf unless there are serious issues with clipping on weapons. I'd also like to ask if weapons will get a weight cap or something? Currently, nothing is stopping someone from building a huge weapon with full blocks and lasers.

**Comment by ProPeach** [comment]
I also hope that it will stay, you don't get all the terrible balancing downsides on a gun as you do if you're clipping systems with a ship. It's a really great way of enabling that extra bit of detail and opportunity to express yourself in the already tiny guns.
On a different note, do you plan on keeping them at their current scale? I think the general consensus on Discord is that the guns are awesome, but could do with being scaled down by a half in terms of size. The output of the weapons is also extremely powerful for something that a player can walk around with, they might also need a nerf

**Comment by tsunamayo** [comment]
No it is a bug - but I guess I could think about it.
Yes I want to add a weight cap, also you are supposed to walk a bit slower. I will also add a feature about accuracy - double handle will make you slower but will get better accuracy.
On the size: I already reduce the size, it is currently at 0.66x. I could lower it even more, at 0.5 it was still okay. I am fine with the big look, also please keep in mind that the end-game for weapon will be to use them with some kind of exosuit when mechs are back - so a bit too big will look better on them I think. 
Yes they are powerfull, if I make them smaller then people would put even more stuff on it ;) . I could scale them down if necessary, but for now it is not based on any gameplay feedback at all - we will see when we have boarding and co...
I will do a suggestion thread and blueprint submission as soon as I deliver a few important features, so we will discuss more in depth.

**Comment by ZachZent** [comment]
This reminded me of my vvveeerrryyy old suggestion. Part 2 of course, 1 is for freeform. I still think this would be a possible solution. Blocks would shrink to the size of bricks when crafting weapons and bricks would shrink proportionally. They are still the same blocks, just shrunk. That way players can make more detailed weapons wile using the current blocks and bricks.
https://github.com/tsunamayo/Starship-EVO/issues/601

**Comment by tsunamayo** [comment]
No I dont think I want to go that way... Many issues, for example I think that would be very complicated to place. Plus it would look out of place! I am fine with simple looking weapon - lets see what you can do with that system first.

**Comment by tsunamayo** [comment]
Plus honestly I have been very impressed by what you guys managed to do with the custom drone - there is so much artistry involved with working that that limitation. Yes they might look better, but I think we would loose a lot of the charm. On the other side I am fine to add more small decorative bricks.

**Comment by ProPeach** [comment]
I like the charm of how they look now, the block/brick idea would make them look *too* detailed, kinda like a Minecraft texture would look out of place if it had a different pixel density to other textures in the game.
People are already making some gorgeous guns thanks to their smaller size and clipping, the only change I'd make is to scale them down a little more.
Can I also mention how excited this makes me for mechs and exo suits? They're going to be incredible and truly unique across the gaming world if they use a similar technology to this

**Comment by tsunamayo** [comment]
you mean a smaller brick size?

**Comment by ProPeach** [comment]
Yes, you mentioned that now it's 66% of a regular brick, in my opinion that's still a little too large. 40% might be better for that
